src/t8_cmesh/t8_cmesh_examples.cxx

```diff
@@ -1955,6 +1955,355 @@ t8_cmesh_new_prism_geometry (sc_MPI_Comm comm)
   return cmesh;
 }
 
+/* Offset-1 is added to each tree_id, this is used in i.e. t8_cmesh_new_disjoint_bricks,
+ * If offset is nonzero, then set_partition must be true and the cmesh is
+ * partitioned and has all trees in conn as local trees.
+ * The offsets on the different processes must add up! */
+static t8_cmesh_t
+t8_cmesh_new_brick_2d_ext (t8_gloidx_t num_x, t8_gloidx_t num_y, int x_periodic, int y_periodic, sc_MPI_Comm comm,
+                           int set_partition, t8_gloidx_t offset)
+{
+  int use_offset;
+  t8_cmesh_t cmesh;
+
+  T8_ASSERT (num_x > 0 && num_y > 0);
+  T8_ASSERT (offset == 0 || set_partition);
+  if (offset) {
+    offset--;
+    use_offset = 1;
+  }
+  else {
+    use_offset = 0;
+  }
+  T8_ASSERT (offset >= 0);
+
+  const int dim = 2;
+  const t8_gloidx_t ntrees = num_x * num_y;
+  const int nverts = 4; /* T8_ECLASS_QUAD */
+
+  t8_cmesh_init (&cmesh);
+  t8_cmesh_register_geometry<t8_geometry_linear> (cmesh, dim);
+
+  /* Setting eclass. */
+  for (t8_gloidx_t itree = 0; itree < ntrees; itree++) {
+    t8_cmesh_set_tree_class (cmesh, itree + offset, T8_ECLASS_QUAD);
+  }
+
+  double vertices[nverts][3];
```

Comment:
You could also use t8_update_box_face_edges and a box to unify the vertex computation for 2 and 3D, just like in hypercube_pad.